compiler/fir/resolve/src/org/jetbrains/kotlin/fir/resolve/transformers/FirStatusResolveTransformer.kt

```diff
@@ -168,11 +174,10 @@ private class FirStatusResolveTransformer(
         return transformDeclaration(typeParameter, data)
     }
 
-    override fun transformBlock(block: FirBlock, data: FirDeclarationStatus?): CompositeTransformResult<FirStatement> {
```

Comment:
Why was this taken out?

Reply:
Because it bails out from the transformation. That is, statements inside a block (in particular, inside the anonymous initializer block) are not transformed by this transformer, if this is still there.

Reply:
It turns out this is intentionally here to stop this transformer from transforming declarations inside block. Reverted.